src/services/spot/Spot.cpp

```diff
@@ -0,0 +1,387 @@
+                   doc.AddMember(value_series_name,outarr,doc.GetAllocator());
+                 }
+              } 
+            }
+            std::ofstream ofs(place.c_str());
+            rapidjson::OStreamWrapper osw(ofs);
+            rapidjson::Writer<rapidjson::OStreamWrapper> writer(osw);
+            doc.Accept(writer);
+          }
+        }
+        // TODO: reimplement
+        Spot()
+            { }
+
+        //
+        // --- callback functions
+        //
+        static std::pair<AggregationHandler,SelectionHandler> create_query_processor(std::string query){
+           CalQLParser parser(query.c_str()); 
+           if (parser.error()) {
+               Log(0).stream() << "spot: config parse error: " << parser.error_msg() << std::endl;
+               return std::make_pair(nullptr,nullptr);
+           }
+           QuerySpec    spec(parser.spec());
+           return std::make_pair(new Aggregator(spec),new RecordSelector(spec));
+        }
+        static std::string query_for_annotation(std::string grouping, std::string metric = "time.inclusive.duration"){
+          std::string end_grouping = "event.end#"+grouping;
+          return "SELECT " + grouping+","+end_grouping+",sum("+metric+") " + "WHERE " +grouping+","+end_grouping+","+metric + " GROUP BY " + end_grouping+","+grouping;
+        }
+        static void pre_write_cb(Caliper* c, const SnapshotRecord* flush_info) {
+            ConfigSet    config(RuntimeConfig::init("spot", s_configdata));
+            const std::string&  config_string = config.get("config").to_string().c_str();
+            include_local = config.get("include_local").to_string().compare("NO");
+            show_exclusive = config.get("show_exclusive").to_string().compare("NO");
+            divisor = config.get("time_divisor").to_int();
+            code_version = config.get("code_version").to_string();
+            recorded_time = config.get("recorded_time").to_string();
+            if(recorded_time.size()==0){
+              time_t rawtime;
+              struct tm * timeinfo;
+              time (&rawtime);
+              timeinfo = localtime (&rawtime);
+              static char time_string[1000];
+              strftime(time_string, sizeof(time_string),"%Y-%m-%d %H:%M:%S", timeinfo);
+              recorded_time =  std::string(time_string);
+            }
+            std::string title_string = config.get("title").to_string();
+            bool use_default_title = (title_string.size() == 0);
+            if(use_default_title){
+              Log(0).stream() << "Spot: using default titles for graphs\n";
+            }
+            std::string y_axes_string = config.get("y_axes").to_string();
+            util::split(y_axes_string,':',std::back_inserter(y_axes));
+            std::vector<std::string> logging_configurations;
+            util::split(config_string,',',std::back_inserter(logging_configurations));
```

Comment:
There's a `.to_stringlist()` now, i.e. `logging_configurations = config.get("config").to_stringlist(',');` should do it.